frontend/src/lib/Quiver.ts

```diff
@@ -111,7 +112,110 @@ type IntervalClosed = "left" | "right" | "both" | "neither"
 
 // Our current Typescript version (3.9.5) doesn't support template literal types,
```

Comment:
nit: Is this comment actually still valid since we are already on typescript version `4.6.3`?

Reply:
Yes. It still applies because we have a newer version of the typescript, but we cannot use its functionality because it is not supported by typescript, so "having" the new version is only theoretical.
```
Prettier......................................................................Failed
- hook id: prettier
- exit code: 2

['./scripts/run_in_subdirectory.py', 'frontend/', 'yarn', 'prettier', '--write', 'frontend/src/lib/Quiver.ts', 'frontend/src/lib/mocks/arrow/types/intervalInt64.ts']
yarn run v1.22.19
$ /Users/kbregula/Snowflake/streamlit/streamlit/frontend/node_modules/.bin/prettier --write src/lib/Quiver.ts src/lib/mocks/arrow/types/intervalInt64.ts
src/lib/Quiver.ts
[error] src/lib/Quiver.ts: SyntaxError: Type expected. (111:21)
[error]   109 | type IntervalData = "int64" | "uint64" | "float64" | "datetime64[ns]"
[error]   110 | type IntervalClosed = "left" | "right" | "both" | "neither"
[error] > 111 | type IntervalType = `interval[${IntervalData}, ${IntervalClosed}]`
[error]       |                     ^
[error]   112 |
[error]   113 | // The frequency strings defined in pandas.
[error]   114 | // See: https://pandas.pydata.org/docs/user_guide/timeseries.html#dateoffset-objects
src/lib/mocks/arrow/types/intervalInt64.ts 106ms
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Reply:
I created PR that updates prettier to latest version: https://github.com/streamlit/streamlit/pull/5455

Reply:
Fixed.